File: Traitement/Annales2/exercice3bis.R
# Définir la fonction exo3
exo3 <- function(filename) {
  # Charger les données
  data <- read.csv(filename, sep = ";")
  
  # 1. Afficher la répartition des indices
  distribution <- table(data$indice)
  print("Répartition des indices :")
  print(distribution)
  
  # 2. Créer un graphique en camembert
  png("r_repartition_indices.png")  # Ouvrir un fichier PNG pour sauvegarder le graphique
  pie(distribution, main = "Répartition des indices")  # Créer le camembert
  dev.off()  # Fermer le fichier PNG
  
  # 3. Afficher la répartition des indices par communes
  commune_distribution <- table(data$commune_nom, data$indice)
  print("Répartition des indices par communes :")
  print(commune_distribution)
  
  # 4. Trouver la commune avec le moins de "bon" qualité de l'air
  min_bon_commune <- names(which.min(table(data$commune_nom[data$indice == "bon"])))
  print("Commune où il y a le moins de bonne qualité de l'air :")
  print(min_bon_commune)
}

# Appeler la fonction exo3 avec le fichier CSV
exo3("Annales2/dataairplqualiteairangers.csv")

File: Traitement/Annales2/exercice3bis.py
import pandas as pd
import matplotlib.pyplot as plt

def exo3(filename):
    # Charger les données
    data = pd.read_csv(filename, delimiter=';')
    
    # Afficher la répartition des indices
    distribution = data['indice'].value_counts()
    print("Répartition des indices :")
    print(distribution)
    
    # Créer un graphique en camembert
    distribution.plot(kind='pie', autopct='%1.1f%%')
    plt.title("Répartition des indices")
    plt.savefig("python_repartition_indices.png")
    plt.close()
    
    # Afficher la répartition des indices par communes
    commune_distribution = data.groupby(['commune_nom', 'indice']).size().unstack()
    print("Répartition des indices par communes :")
    print(commune_distribution)
    
    # Trouver la commune avec le moins de "bon" qualité de l'air
    min_bon_commune = data[data['indice'] == 'bon']['commune_nom'].value_counts().idxmin()
    print("Commune où il y a le moins de bonne qualité de l'air :")
    print(min_bon_commune)

if __name__ == "__main__":
    exo3("Annales2/dataairplqualiteairangers.csv")